crates/core/src/db/queries/metadatas.rs

```diff
@@ -264,23 +269,32 @@ pub fn list(
         );
     }
 
-    if let Some(offerers) = offerers {
+    if offerers.is_some() || (with_offers.is_some() && with_offers.unwrap_or(false)) {
```

Comment:
```suggestion
    if offerers.is_some() || with_offers.is_some() {
```
Since LN285 only matches on true conditions logic is easier to follow with just the OR condition.

https://github.com/holaplex/indexer/pull/510/files#diff-9a080dfe669d746b43c043fc557a06d5add886b23833af6daa7d80745c432520R285

Reply:
With your change, if `offerers` is `None` and `with_offers` is `false`, then this block will execute, adding query conditions that are not actually used for anything. You could argue that since the current version of the query will fail if `with_offers` is `false`, then that condition could not be met. However, that seems like an antipattern to me, to make an assumption about the value of `with_offers` based on the code reaching this point.

Reply:
block the execution? The query doesnt fail if with_offers is false it just removes nfts that don't have offers which isn't the normal usage pattern but is a valid query condition.

Reply:
Ok is see only matching on true then unwrap with_offers before and then references bellow.

```rust
let with_offers = with_offers.unwrap_or(false); // not that user can still pass false to the option which will just skip adding the filter bc its false

if offerers.is_some() || with_offers {
  ...
  
  if with_offers {
    // add extract bid filter
  }
}
```